### PR TITLE
update the tidb configuration document

### PR DESCRIPTION
**Pull request by cfzjywxk**
### What is changed, added or deleted? (Required)

Update the `tidb` config file and command-line argument documents, add new configs and remove deprecated ones 

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

**Comment by TomShawn** [comment]
@cfzjywxk Because this week is Document Special Week, the merge button is temporarily disabled, so we can only merge this PR until the end of the Special Week (presumably this Friday night). Is it acceptable for you?

File: tidb-configuration-file.md [hunk 1]
```diff
@@ -528,7 +529,7 @@ TiDB 服务状态相关配置。
 
 ### max-retry-count
 
-+ 悲观事务中每个语句最大重试次数，超出该限制将会报错。
++ 悲观事务中每个语句最大重试次数，超出该限该语句执行制将会报错。
```
Comment (coocood) [inline tidb-configuration-file.md:532]: `该限该语句`？

File: tidb-configuration-file.md [hunk 2]
```diff
@@ -270,19 +270,19 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 
 ### `max-memory`
 
-+ Prepare cache LRU 使用的最大内存限制，超过 performance.max-memory * (1 - prepared-plan-cache.memory-guard-ratio) 会剔除 LRU 中的元素。
++ Prepare plan cache LRU 使用的最大内存限制，超过 performance.max-memory * (1 - prepared-plan-cache.memory-guard-ratio) 会剔除 LRU 中的元素。
```
Comment (coocood) [inline tidb-configuration-file.md:273]: It's `max-memory`. it has nothing to do with `plan cache`.

Comment (coocood) [inline tidb-configuration-file.md:273]: So the comment for this option in the config template is wrong.

Comment (coocood) [reply tidb-configuration-file.md:273]: https://github.com/pingcap/tidb/blob/master/config/config.toml.example#L200

File: tidb-configuration-file.md [hunk 3]
```diff
@@ -391,11 +391,12 @@ prepare 语句的 Plan cache 设置。
 + 默认值：41s
 + 这个值必须是大于两倍 Raft 选举的超时时间。
 
-### `max-txn-time-use`
+### `max-txn-ttl`
 
-+ 单个事务允许的最大执行时间。
-+ 默认值：590
-+ 单位：秒
++ 单个事务允许持有相关资源的最大执行时间。
```
Comment (coocood) [inline tidb-configuration-file.md:396]: `持有相关资源` 不太容易理解

File: tidb-configuration-file.md (hunk 2 above)
Comment (cfzjywxk) [reply tidb-configuration-file.md:273]: Seems this variable is used by them both, https://github.com/pingcap/tidb/blob/master/tidb-server/main.go#L593.
Maybe we should change the variable name to distinguish them?

File: tidb-configuration-file.md (hunk 3 above)
Comment (cfzjywxk) [reply tidb-configuration-file.md:396]: `单个事务有效存活时间，超过该时间事务资源可能被释放`  
这样可否？

File: tidb-configuration-file.md (hunk 2 above)
Comment (coocood) [reply tidb-configuration-file.md:273]: We have prepared plan cache capacity already.
Maybe we don't need this value for prepared plan cache.

File: tidb-configuration-file.md (hunk 3 above)
Comment (coocood) [reply tidb-configuration-file.md:396]: `单个事务持锁的最长时间, 超过该时间，该事务的锁可能会被其他事务清除，导致该事务无法成功执行。`

**Comment by coocood** [comment]
LGTM

File: tidb-configuration-file.md [hunk 4]
```diff
@@ -270,19 +270,19 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 
 ### `max-memory`
 
-+ Prepare cache LRU 使用的最大内存限制，超过 performance.max-memory * (1 - prepared-plan-cache.memory-guard-ratio) 会剔除 LRU 中的元素。
++ TiDB 最大使用内存，单位字节
```
Comment (zz-jason) [inline tidb-configuration-file.md:273]:
```suggestion
+ TiDB 最大使用内存，单位：字节
```

File: command-line-flags-for-tidb-configuration.md [hunk 5]
```diff
@@ -37,6 +37,11 @@ aliases: ['/docs-cn/dev/reference/configuration/tidb-server/configuration/']
 + 默认："0.0.0.0"
 + 0.0.0.0 默认会监听所有的网卡地址。如果有多块网卡，可以指定对外提供服务的网卡，如 192.168.100.113
 
+## `enable-binlog`
+
++ 是否产生 tidb binlog
```
Comment (yikeke) [inline command-line-flags-for-tidb-configuration.md:42]:
```suggestion
+ 是否产生 TiDB binlog
```

File: command-line-flags-for-tidb-configuration.md [hunk 6]
```diff
@@ -37,6 +37,11 @@ aliases: ['/docs-cn/dev/reference/configuration/tidb-server/configuration/']
...
++ 默认: false
```
Comment (yikeke) [inline command-line-flags-for-tidb-configuration.md:43]:
```suggestion
+ 默认：false
```
s/默认:/默认：

其他地方也批量替换下吧

File: command-line-flags-for-tidb-configuration.md [hunk 7]
```diff
@@ -149,3 +154,33 @@ aliases: ['/docs-cn/dev/reference/configuration/tidb-server/configuration/']
...
++ 默认: "/data/deploy/plugin"
+
+## `--plugin-load`
+
++ 需要加载的 plugin 列表，以 "," 逗号分隔
++ 默认: ""
+
+## `--affinity-cpus`
+ 
++ 设置 TiDB server cpu 亲和性，以 "," 逗号分隔，例如 "1,2,3"
```
Comment (yikeke) [inline command-line-flags-for-tidb-configuration.md:170]:
```suggestion
+ 设置 TiDB server CPU 亲和性，以 "," 逗号分隔，例如 "1,2,3"
```

File: command-line-flags-for-tidb-configuration.md [hunk 8]
```diff
@@ -149,3 +154,33 @@ aliases: ['/docs-cn/dev/reference/configuration/tidb-server/configuration/']
...
++ 默认: false
+
+## `--repair-list`
+
++ 修复模式下需要修复的表名
++ 默认: ""
+
+## `--require-secure-transport`
+
++ 是否要求客户端使用 secure 传输模式
```
Comment (yikeke) [inline command-line-flags-for-tidb-configuration.md:185]:
```suggestion
+ 是否要求客户端使用安全传输模式
```

File: command-line-flags-for-tidb-configuration.md [hunk 9]
```diff
@@ -37,6 +37,11 @@ aliases: ['/docs-cn/dev/reference/configuration/tidb-server/configuration/']
 + 默认："0.0.0.0"
 + 0.0.0.0 默认会监听所有的网卡地址。如果有多块网卡，可以指定对外提供服务的网卡，如 192.168.100.113
 
+## `enable-binlog`
```
Comment (pcqz) [inline command-line-flags-for-tidb-configuration.md:40]:
```suggestion
## `--enable-binlog`
```

File: command-line-flags-for-tidb-configuration.md [hunk 10]
```diff
@@ -149,3 +154,33 @@ aliases: ['/docs-cn/dev/reference/configuration/tidb-server/configuration/']
...
 + 默认：""
+
+## `--plugin-dir`
+
++ plugin 存放目录
++ 默认："/data/deploy/plugin"
+
+## `--plugin-load`
+
++ 需要加载的 plugin 列表，以 "," 逗号分隔
```
Comment (pcqz) [inline command-line-flags-for-tidb-configuration.md:165]:
```suggestion
+ 需要加载的 plugin 名称，多个 plugin 以 "," 逗号分隔
```

File: tidb-configuration-file.md [hunk 11]
```diff
@@ -168,6 +173,12 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 + 默认值：300ms
 + 当查询大于这个值，就会当做是一个慢查询，输出到慢查询日志。
 
+### `record-plan-in-slow-log`
+
++ 在慢日志中记录执行计划
++ 默认值：1
++ 0 表示关闭，1 表示开启，默认开启
+
```
Comment (pcqz) [inline tidb-configuration-file.md:181]:
```suggestion
该值作为系统变量 tidb_record_plan_in_slow_log 的初始值。
```

**Comment by pcqz** [comment]
LGTM

File: tidb-configuration-file.md (hunk 11 above)
Comment (yikeke) [reply tidb-configuration-file.md:181]: 建议这儿给个 tidb_record_plan_in_slow_log 参数的文档链接，如果没有这个参数描述的话，需要补一下 @cfzjywxk

File: tidb-configuration-file.md [hunk 12]
```diff
@@ -177,7 +177,7 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 
 + 在慢日志中记录执行计划
 + 默认值：1
-+ 0 表示关闭，1 表示开启，默认开启
++ 0 表示关闭，1 表示开启，默认开启，该值作为系统变量 tidb_record_plan_in_slow_log 的初始值。
```
Comment (yikeke) [inline tidb-configuration-file.md:180]:
```suggestion
+ 0 表示关闭，1 表示开启，默认开启。该配置项的值会作为系统变量 tidb_record_plan_in_slow_log 的初始值。
```

File: tidb-configuration-file.md (hunk 11 above)
Comment (yikeke) [reply tidb-configuration-file.md:181]: ping @cfzjywxk

Comment (cfzjywxk) [reply tidb-configuration-file.md:181]: Got

File: tidb-configuration-file.md (hunk 2 above)
Comment (cfzjywxk) [reply tidb-configuration-file.md]: Related [tidb-issue](https://github.com/pingcap/tidb/issues/17298)

Comment (SunRunAway) [reply tidb-configuration-file.md]: > Fix the annotation in tidb.toml.example and tell the readers that max-memory is only for Plan Cache in tidb.toml.example in all release branches(v2.1, v3.0, v3.1, v4.0).

The [issue](https://github.com/pingcap/tidb/issues/17298#issuecomment-632180632) said that `max-memory` is only for Plan Cache, we should change this documentation back.

Comment (SunRunAway) [reply tidb-configuration-file.md]: Fixed in https://github.com/pingcap/docs-cn/pull/3421

**Pull request merged**